mewarpx/mewarpx/diags_store/flux_diagnostic.py

```diff
@@ -667,13 +667,39 @@ def __init__(self, diag_steps, runinfo, overwrite=True,
             **kwargs
         )
 
+        self.check_scraping()
+
         # if this run is from a restart, try to load flux data up to the
         # current point
-        if mwxrun.restart:
-            self._load_checkpoint_flux()
+        # mwxrun.restart isn't set until mwxrun.init_run is called, so the
+        # lambda function is needed here
+        callbacks.installafterinit(
+            lambda _=None: self._load_checkpoint_flux() if mwxrun.restart else None
+        )
 
         callbacks.installafterstep(self._flux_ana)
 
+    def check_scraping(self):
+        """Checks that particles scraping at the domain boundaries is turned on
+        when flux diagnostics are on."""
+        scraper_labels = []
+        for surface in self.surface_dict.values():
+            scraper_labels.append(surface.scraper_label)
+
+        for spec in mwxrun.simulation.species:
+            if "zhi" in scraper_labels:
+                if not hasattr(spec, "save_particles_at_zhi"):
+                    raise ValueError(f"zhi scraping is turned off for "
+                                     f"{spec.name} and there is an anode.")
+            if "zlo" in scraper_labels:
+                if not hasattr(spec, "save_particles_at_zlo"):
+                    raise ValueError(f"zlo scraping is turned off for "
+                                     f"{spec.name} and there is a cathode.")
+            if "eb" in scraper_labels:
+                if not hasattr(spec, "save_particles_at_eb"):
+                    raise ValueError(f"eb scraping is turned off for "
+                                     f"{spec.name} and there is an EB.")
+
```

Comment:
```suggestion
        for spec in mwxrun.simulation.species:
            for surfname, surface in self.surface_dict.items():
                if not hasattr(spec, f"save_particles_at_{surface.scraper_label}"):
                    raise ValueError(f"{surface.scraper_label} scraping is turned off "
                                     f"for {spec.name} but {surfname} requires it.")

```

Reply:
I think that works and is more concise? But you should test it and fix line breaks if I messed those up.

Reply:
I think there will be a slight difficulty here in that for example the cathode has `scraper_label = 'z_lo'` but scraping from the cathode requires `save_particles_at_zlo`. So there will be a need to remove the `_`.